lib/broadway_cloud_pub_sub/producer.ex

```diff
@@ -153,7 +175,9 @@ defmodule BroadwayCloudPubSub.Producer do
 
     {specs, opts} = prepare_to_connect(broadway_opts[:name], client, opts)
 
-    :persistent_term.put(ack_ref, Map.new(opts))
+    ack_opts = Keyword.put(opts, :topology_name, broadway_opts[:name])
+
+    :persistent_term.put(ack_ref, Map.new(ack_opts))
```

Comment:
The `ack_ref` itself is the topology name, so I think we can keep the previous one:

```suggestion
    :persistent_term.put(ack_ref, Map.new(opts))
```

Reply:
Ignore. :)